[Bug]test_single_decode_torch_compile_cuda_graph fails due to version check in subprocess

Found this issue in nightly public CI. First found this issue in [Nightly Release #220](https://github.com/flashinfer-ai/flashinfer/actions/runs/24273042010)
Failed job example: [https://github.com/flashinfer-ai/flashinfer/actions/runs/24273042010/job/70890382534](https://github.com/flashinfer-ai/flashinfer/actions/runs/24273042010/job/70890382534)
```E       AssertionError: Test failed:
E         stdout: 
E         stderr: _env
E           File "/workspace/flashinfer/jit/env.py", line 94, in <module>
E             FLASHINFER_CUBIN_DIR: pathlib.Path = _get_cubin_dir()
E                                                  ^^^^^^^^^^^^^^^^
E           File "/workspace/flashinfer/jit/env.py", line 76, in _get_cubin_dir
E             raise RuntimeError(
E         RuntimeError: flashinfer-cubin version (0.6.7.dev20260411) does not match flashinfer version (0.0.0+unknown). Please install the same version of both packages. Set FLASHINFER_DISABLE_VERSION_CHECK=1 to bypass this check.
E         
E       assert (1 == 0)
E        +  where 1 = CompletedProcess(args=['/opt/conda/envs/py312/bin/python3.12', '-c', 'import torch\nfrom flashinfer import single_deco...wn). Please install the same version of both packages. Set FLASHINFER_DISABLE_VERSION_CHECK=1 to bypass this check.\n').returncode

tests/attention/test_batch_decode_kernels.py:824: AssertionError
=========================== short test summary info ============================
FAILED tests/attention/test_batch_decode_kernels.py::test_single_decode_torch_compile_cuda_graph
```

This may relate to PR [#2857](https://github.com/flashinfer-ai/flashinfer/pull/2857) which added test_single_decode_torch_compile_cuda_graph as a regression test for issue #541. The test runs a subprocess  via subprocess.run() to work around torch.compile's dynamo tracing restrictions.

 Inside the subprocess, importing flashinfer triggers the module-level version check in flashinfer/jit/env.py:94:
``` FLASHINFER_CUBIN_DIR: pathlib.Path = _get_cubin_dir()```

  In CI, flashinfer is installed from source without generating _build_meta.py, so flashinfer.__version__ falls back to "0.0.0+unknown". Meanwhile flashinfer-cubin is installed as a versioned nightly package (0.6.7.dev20260411). The strict equality check at line 74 raises:

```RuntimeError: flashinfer-cubin version (0.6.7.dev20260411) does not match flashinfer version (0.0.0+unknown).```

  The test never reaches the actual torch.compile / CUDA graph logic — it fails at import time.The version check itself does not handle the "0.0.0+unknown" case, which is the standard fallback for editable/source installs without build metadata.

--- a/flashinfer/jit/env.py
+++ b/flashinfer/jit/env.py
@@ -69,8 +69,11 @@ def _get_cubin_dir():
 
         flashinfer_cubin_version = flashinfer_cubin.__version__
         # Allow bypassing version check with environment variable
+        # NOTE(yiyang): skip version check for editable/source installs where
+        # flashinfer_version falls back to "0.0.0+unknown" (no _build_meta.py).
         if (
             not os.getenv("FLASHINFER_DISABLE_VERSION_CHECK")
+            and flashinfer_version != "0.0.0+unknown"
             and flashinfer_version != flashinfer_cubin_version
         ):
             raise RuntimeError(
@@ -108,9 +111,11 @@ def _get_aot_dir():
         # NOTE(Zihao): we don't use exact version match here because the version of flashinfer-jit-cache
         # contains the CUDA version suffix: e.g. 0.3.1+cu129.
         # Allow bypassing version check with environment variable
-        if not os.getenv(
-            "FLASHINFER_DISABLE_VERSION_CHECK"
-        ) and not flashinfer_jit_cache_version.startswith(flashinfer_version):
+        if (
+            not os.getenv("FLASHINFER_DISABLE_VERSION_CHECK")
+            and flashinfer_version != "0.0.0+unknown"
+            and not flashinfer_jit_cache_version.startswith(flashinfer_version)
+        ):
             raise RuntimeError(
                 f"flashinfer-jit-cache version ({flashinfer_jit_cache_version}) does not match "
                 f"flashinfer version ({flashinfer_version}). "

